src/utils/pluginConfigGenerators.js

```diff
@@ -0,0 +1,97 @@
+const path = require('path');
+
+
+/**
+ * Find main version of a plugin by resolving it to the default version or the first version configured.
+ * @param {import('../common/components/Switcher').Doc} plugin
+ */
+function findMainVersion(plugin) {
+  let mainVersion = plugin.versions[0];
+  if (plugin.defaultVersion) {
+    const foundVersion = plugin.versions.find(
+      (version) => version.label === plugin.defaultVersion,
+    );
+    if (!foundVersion)
+      throw `Default version ${plugin.defaultVersion} of doc ${plugin.label} not found.`;
+
+    mainVersion = foundVersion;
+  }
+
+  return mainVersion;
+}
+
+/**
+ * Generate the plugin config from the versioned config.
+ * @param {import('../common/components/Switcher').Doc[]} pluginConfig
+ * @param {string} basePath
+ */
+function generatePluginConfig(pluginConfig, basePath) {
+  let plugins = [];
+
+  for (const plugin of pluginConfig) {
+    const mainVersion = findMainVersion(plugin);
+
+    for (const version of plugin.versions) {
+      const { label, badges, ...rest } = version;
+      // TODO: This could be removed once we don't use points in paths anymore.
+      const plugin_name_path = plugin.routeBasePath
+        ? plugin.routeBasePath
+        : plugin.id;
+      const extended_base_path =
+        basePath + plugin_name_path + '/' + (label ? label : '');
+      if (plugin.versions.length > 1) {
+        plugins.push({
+          id: plugin.id + (label ? '-' + label.replace(/\./g, '-') : ''),
+          path: path.resolve(extended_base_path + '/docs'),
+          routeBasePath: plugin_name_path,
+          sidebarPath: path.resolve(extended_base_path + '/sidebars.js'),
+          versions:{
+                  current: {
+                    label: label,
+                    path: mainVersion.label === label ? undefined : label,
+                    badge: true,
+                  },
+          },
+          ...rest,
+        });
+      }
+      else {
+        plugins.push({
+          id: plugin.id + (label ? '-' + label.replace(/\./g, '-') : ''),
+          path: path.resolve(extended_base_path + '/docs'),
+          routeBasePath: plugin_name_path,
+          sidebarPath: path.resolve(extended_base_path + '/sidebars.js'),
+          ...rest,
+        });
+      }
+    }
+  }
+
+  return plugins;
+}
```

Comment:
Just a suggestion to make this a bit shorter. wdyt?

```suggestion
function generatePluginConfig(pluginConfig, basePath) {
  return pluginConfig.flatMap((doc) => {
    const mainVersion = findMainVersion(doc);

    return doc.versions.map((version) => {
      const { label, badges, ...rest } = version;

      // TODO: This could be removed once we don't use points in paths anymore.
      const plugin_name_path = doc.routeBasePath ? doc.routeBasePath : doc.id;

      const extended_base_path =
        basePath + plugin_name_path + '/' + (label ? label : '');

      plugin = {
        id: doc.id + (label ? '-' + label.replace(/\./g, '-') : ''),
        path: path.resolve(extended_base_path + '/docs'),
        routeBasePath: plugin_name_path,
        sidebarPath: path.resolve(extended_base_path + '/sidebars.js'),
        ...rest,
      };

      if (doc.versions.length > 1) {
        plugin.versions = {
          current: {
            label,
            path: mainVersion.label === label ? undefined : label,
            badge: true,
          },
        };
      }

      return plugin;
    });
  });
}
```

Reply:
I just realised I didn't push everything 😆.
But yeah, will apply your changes. Makes sense